diff --git a/src/shared/linux_osl.c b/src/shared/linux_osl.c
index 6157d18..15f64cb 100644
--- a/src/shared/linux_osl.c
+++ b/src/shared/linux_osl.c
@@ -1076,7 +1076,7 @@ osl_os_get_image_block(char *buf, int len, void *image)
 	if (!image)
 		return 0;
 
-	rdlen = kernel_read(fp, fp->f_pos, buf, len);
+	rdlen = kernel_read(fp, buf, len, &fp->f_pos);
 	if (rdlen > 0)
 		fp->f_pos += rdlen;
 
diff --git a/src/wl/sys/wl_linux.c b/src/wl/sys/wl_linux.c
index 0d05100..6f9b362 100644
--- a/src/wl/sys/wl_linux.c
+++ b/src/wl/sys/wl_linux.c
@@ -769,11 +769,13 @@ wl_pci_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 	pci_set_master(pdev);
 
 	pci_read_config_dword(pdev, 0x40, &val);
-	if ((val & 0x0000ff00) != 0)
+	if ((val & 0x0000ff00) != 0) {
+		// FIXME is this all supposed to be in a block? indended like so, but not blocked
 		pci_write_config_dword(pdev, 0x40, val & 0xffff00ff);
 		bar1_size = pci_resource_len(pdev, 2);
 		bar1_addr = (uchar *)ioremap_nocache(pci_resource_start(pdev, 2),
 			bar1_size);
+	}
 	wl = wl_attach(pdev->vendor, pdev->device, pci_resource_start(pdev, 0), PCI_BUS, pdev,
 		pdev->irq, bar1_addr, bar1_size);
 
@@ -2053,8 +2055,12 @@ wl_osl_pcie_rc(struct wl_info *wl, uint op, int param)
 void
 wl_dump_ver(wl_info_t *wl, struct bcmstrbuf *b)
 {
+#if defined(BCMDBG)
 	bcm_bprintf(b, "wl%d: %s %s version %s\n", wl->pub->unit,
 		__DATE__, __TIME__, EPI_VERSION_STR);
+#else
+	bcm_bprintf(b, "wl%d: version %s\n", wl->pub->unit, EPI_VERSION_STR);
+#endif
 }
 
 #if defined(BCMDBG)
@@ -2954,8 +2960,9 @@ _wl_add_monitor_if(wl_task_t *task)
 		goto done;
 	}
 
-	ASSERT(strlen(wlif->name) > 0);
-	strncpy(wlif->dev->name, wlif->name, strlen(wlif->name));
+	ASSERT(strnlen(wlif->name, sizeof(wlif->name)) > 0);
+	strncpy(wlif->dev->name, wlif->name, sizeof(wlif->dev->name));
+
 
 	wl->monitor_dev = dev;
 	if (wl->monitor_type == 1)
